### PR TITLE
Projects - Garage improvements

### LINKED ISSUE #2138: [Change Request]: Adapt Info Section of Hakon's Garage System
### Is the requested change related to a problem?

The Info section is a bit unstructured for the vehicles. 

### Describe the solution you'd like.

1. Suggest to do add seperation for "general information about the garage" and "Vehicle Specific Information"

2. Suggest to add a summarized Percentage Indicator for each, Ammo / Fuel / Repair of the selected Vehicle. 

If possible, 
- add a color coding towards these. ( 0% Red15% Yellow 50% White 100% )
- if vehicle with no Weaponary, display "Ammo -"

### Alternative solutions

_No response_

### Additional context

MockUp:

### PR DESCRIPTION
**Pull request by HakonRydland**
## What type of PR is this.
1. [ ] Bug
2. [x] Change
3. [x] Enhancement

### What have you changed and why?
Information:
    - Added synergy with fuel scarcity with actual fuel consumption on un-garage (vanilla still practically still has unlimited fuel due to vanilla fuel truck configs).
    - Added long time requested QOL changes #2138 #2006 
    - Added alphabetical sorting to the vehicle lists
    - Fixed a type causing an error in capacity check code
    - Added a wrapper around the info panel to aid with extending text OOB

### Please specify which Issue this PR Resolves.
closes #2138 #2006 

### Please verify the following and ensure all checks are completed.

1. [x] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [x] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps: 

********************************************************
Notes:

### CODE REVIEW

File: A3-Antistasi/Garage/Refuel/fn_reciveStateUpdate.sqf [hunk 1]
```diff
@@ -0,0 +1,6 @@
+params ["_vUID","_stateIndex", "_state"];
```
[the remark this answers is not in the document]
Comment (killerswin2) [reply A3-Antistasi/Garage/Refuel/fn_reciveStateUpdate.sqf:1]: Missing your defines here.

File: A3-Antistasi/Garage/Refuel/fn_refuelVehicleFromSources.sqf [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            _fuelData set [1, (_fuelCargo - _neededCapacity) / _transportFuel]; //vanilla stores fuel cargo as percentage
+        };
+        _neededCapacity = 0;
+    };
+    _stateChanges pushBack [_sourceUID, _fuelData, 1];
+};
+
+//broadcast source vehicle changes to clinets
+{ _x call HR_GRG_fnc_broadcastStateUpdate; } forEach _stateChanges;
+if (_sourceEmptied) then { [1] call HR_GRG_fnc_declairSources; };
```
Comment (killerswin2) [inline A3-Antistasi/Garage/Refuel/fn_refuelVehicleFromSources.sqf:73]: misspelling HR_GRG_fnc_declairSources -> HR_GRG_fnc_declareSources

File: A3-Antistasi/Garage/Extras/fn_reloadExtras.sqf [hunk 3]
```diff
@@ -154,6 +155,63 @@ private _typeSource = switch (_source find true) do {
...
+        private _maxAmmo = getNumber (configFile/"CfgMagazines"/_mag/"count");
+        _sumPercent = _sumPercent + (_count/_maxAmmo);
+    } forEach _this;
+    _sumPercent / count _this;
+};
+
+private _hasAmmo = (HR_GRG_previewVehState#2) isNotEqualTo [];
+private _avgAmmo = (HR_GRG_previewVehState#2) call _getPrecentageAmmo;
+private _avgFuel = HR_GRG_previewVehState#0#0;
+private _avgDmg = 1 - (HR_GRG_previewVehState#1#0);
```
Comment (killerswin2) [inline A3-Antistasi/Garage/Extras/fn_reloadExtras.sqf:179]: the states, could be abstracted to some define?

File: A3-Antistasi/Garage/Refuel/fn_refuelVehicleFromSources.sqf [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+private _sourceEmptied = false;
+private _stateChanges = [];
+while {count (HR_GRG_Sources#1) > 0} do {
+    if (_neededCapacity == 0) exitWith {};
+    Trace_1("Needed capacity: %1", _neededCapacity);
+
+    private _sourceUID = HR_GRG_Sources#1#0;
+    private "_sourceData";
+    for "_i" from 0 to 4 do { _sourceData = (HR_GRG_Vehicles#_i) get _sourceUID; if (!isNil "_sourceData") exitWith {}; }; //find vehicles in categorys, typically cat 0 "cars"
+    private _fuelData = _sourceData#4#0; //vehicle data >> statePreservation data >> Fuel data
```
[the remark this answers is not in the document]
Comment (killerswin2) [reply A3-Antistasi/Garage/Refuel/fn_refuelVehicleFromSources.sqf:46]: #define statedata 4
#define fuelState 0
#define aceFuelCargo 2 
#define fuelCargo 0
#define vehFuel 0
or ignore do to dm talk

**Review by killerswin2 — CHANGES_REQUESTED** [review]

File: A3-Antistasi/Garage/Refuel/fn_refuelVehicleFromSources.sqf (hunk 4 above)
[the remark this answers is not in the document]
Comment (CalebSerafin) [reply A3-Antistasi/Garage/Refuel/fn_refuelVehicleFromSources.sqf:46]: So I've personally become laxed on the magic array indices. The reason is that SQF has not compiled structures or classes where property names can be used. hashMaps can be an unnecessary overhead for being a small container. Although, for arrays 4 or longer I do generally use Macros to name indices. But it is a good suggestion.

**Review by CalebSerafin — COMMENTED** [review]
Yeah, I think killer caught all the critical points.

File: A3-Antistasi/Garage/Refuel/fn_refuelVehicleFromSources.sqf (hunk 4 above)
[the remark this answers is not in the document]
Comment (HakonRydland) [reply A3-Antistasi/Garage/Refuel/fn_refuelVehicleFromSources.sqf:46]: i feel the comments are good enough

File: A3-Antistasi/Garage/Extras/fn_reloadExtras.sqf (hunk 3 above)
Comment (HakonRydland) [reply A3-Antistasi/Garage/Extras/fn_reloadExtras.sqf:179]: comments should help future readers easily understand

File: A3-Antistasi/Garage/Refuel/fn_reciveStateUpdate.sqf (hunk 1 above)
[the remark this answers is not in the document]
Comment (HakonRydland) [reply A3-Antistasi/Garage/Refuel/fn_reciveStateUpdate.sqf:1]: comment explains the "magic numbers"

**Review by killerswin2 — APPROVED** [review]
Good on my end, now Frost.

**Pull request merged**